### ISSUE TITLE
Passing ReadFile result to CSV errors

### ISSUE THREAD

**Issue by matbesancon** [fhs/ZipFile.jl#81]
If the error is linked to CSV, feel free to let me know and we can transfer the issue:

```julia
# download movielens data
using ZipFile, CSV, DataFrames

temp_zipfile = download("http://files.grouplens.org/datasets/movielens/ml-latest-small.zip")

zarchive = ZipFile.Reader(temp_zipfile)

movies_file = zarchive.files[findfirst(f -> occursin("movies", f.name), zarchive.files)]
movies_frame = CSV.read(movies_file, DataFrame)
```
Resulting in the following error at the last line:
```
ERROR: LoadError: MethodError: no method matching readavailable(::ZipFile.ReadableFile)
Closest candidates are:
  readavailable(::Base.AbstractPipe) at io.jl:380
  readavailable(::Base.GenericIOBuffer) at iobuffer.jl:470
  readavailable(::IOStream) at iostream.jl:379
  ...
Stacktrace:
  [1] write
    @ ./io.jl:716 [inlined]
  [2] buffer_to_tempfile
    @ ~/.julia/packages/CSV/eB98W/src/utils.jl:215 [inlined]
  [3] getbytebuffer(x::ZipFile.ReadableFile, buffer_in_memory::Bool)
    @ CSV ~/.julia/packages/CSV/eB98W/src/utils.jl:178
  [4] getsource(x::Any, buffer_in_memory::Bool)
    @ CSV ~/.julia/packages/CSV/eB98W/src/utils.jl:198
  [5] CSV.Context(source::CSV.Arg, header::CSV.Arg, normalizenames::CSV.Arg, datarow::CSV.Arg, skipto::CSV.Arg, footerskip::CSV.Arg, transpose::CSV.Arg, comment::CSV.Arg, ignoreemptyrows::CSV.Arg, ignoreemptylines::CSV.Arg, select::CSV.Arg, drop::CSV.Arg, limit::CSV.Arg, buffer_in_memory::CSV.Arg, threaded::CSV.Arg, ntasks::CSV.Arg, tasks::CSV.Arg, rows_to_check::CSV.Arg, lines_to_check::CSV.Arg, missingstrings::CSV.Arg, missingstring::CSV.Arg, delim::CSV.Arg, ignorerepeated::CSV.Arg, quoted::CSV.Arg, quotechar::CSV.Arg, openquotechar::CSV.Arg, closequotechar::CSV.Arg, escapechar::CSV.Arg, dateformat::CSV.Arg, dateformats::CSV.Arg, decimal::CSV.Arg, truestrings::CSV.Arg, falsestrings::CSV.Arg, type::CSV.Arg, types::CSV.Arg, typemap::CSV.Arg, pool::CSV.Arg, downcast::CSV.Arg, lazystrings::CSV.Arg, stringtype::CSV.Arg, strict::CSV.Arg, silencewarnings::CSV.Arg, maxwarnings::CSV.Arg, debug::CSV.Arg, parsingdebug::CSV.Arg, streaming::CSV.Arg)
    @ CSV ~/.julia/packages/CSV/eB98W/src/context.jl:320
  [6] #File#23
    @ ~/.julia/packages/CSV/eB98W/src/file.jl:218 [inlined]
  [7] File
    @ ~/.julia/packages/CSV/eB98W/src/file.jl:218 [inlined]
  [8] read(source::ZipFile.ReadableFile, sink::Type; copycols::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ CSV ~/.julia/packages/CSV/eB98W/src/CSV.jl:66
  [9] read(source::ZipFile.ReadableFile, sink::Type)
    @ CSV ~/.julia/packages/CSV/eB98W/src/CSV.jl:63
 [10] top-level scope
    @ ~/Documents/frank_wolfe/FrankWolfe.jl/examples/movielens.jl:21
 [11] include(fname::String)
    @ Base.MainInclude ./client.jl:444
 [12] top-level scope
    @ REPL[3]:1
in expression starting at /home/mbesancon/Documents/frank_wolfe/FrankWolfe.jl/examples/movielens.jl:21
```

**Comment by matbesancon** [comment]
@fhs this seems to have been resolved in the last commit, could we have a new version to get the fix? :)

**Comment by fhs** [comment]
Fixed by 3493e0030057b4c5d2c84e5b23ff872e9daa33a2. Closing.

**Comment by matbesancon** [comment]
thank you!
